### PR TITLE
Resolve MaterialApp.locale, WidgetsApp.locale per supportedLocales

### LINKED ISSUE #19551: Assertion failure of localization in BottomNavigationBar
```
Here is the stacktrace: flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown building BottomNavigationBar(dirty, state:
flutter: _BottomNavigationBarState#f2150(tickers: tracking 5 tickers)):
flutter: 'package:flutter/src/material/bottom_navigation_bar.dart': Failed assertion: line 403 pos 12:
flutter: 'localizations != null': is not true.
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      _BottomNavigationBarState._createTiles (package:flutter/src/material/bottom_navigation_bar.dart)
flutter: #3      _BottomNavigationBarState.build (package:flutter/src/material/bottom_navigation_bar.dart:517:47)
flutter: #4      StatefulElement.build (package:flutter/src/widgets/framework.dart:3743:27)
flutter: #5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3655:15)
flutter: #6      Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #7      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #8      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #9      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #10     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #11     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4772:32)
flutter: #12     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #13     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #14     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #15     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #16     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #17     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #18     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #19     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #20     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #21     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #22     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #23     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #24     ParentDataElement.mount (package:flutter/src/widgets/framework.dart:3968:11)
flutter: #25     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #26     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4772:32)
flutter: #27     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #28     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #29     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #30     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #31     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #32     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #33     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #34     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #35     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #36     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #37     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #38     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #39     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #40     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #41     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #42     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #43     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #44     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #45     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #46     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #47     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #48     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #49     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #50     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #51     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #52     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #53     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #54     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #55     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #56     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #57     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #58     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #59     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #60     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #61     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #62     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #63     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #64     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #65     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #66     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #67     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #68     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #69     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #70     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #71     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #72     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #73     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #74     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #75     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #76     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #77     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #78     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #79     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #80     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #81     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #82     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #83     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #84     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #85     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #86     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #87     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #88     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #89     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #90     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #91     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #92     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #93     Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #94     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #95     Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #96     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #97     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #98     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #99     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #100    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #101    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #102    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #103    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #104    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #105    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #106    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #107    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #108    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #109    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #110    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #111    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #112    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #113    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #114    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #115    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4772:32)
flutter: #116    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #117    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #118    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #119    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #120    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #121    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #122    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #123    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #124    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #125    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #126    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #127    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #128    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #129    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #130    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #131    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #132    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #133    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #134    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #135    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #136    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #137    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #138    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #139    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #140    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #141    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #142    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #143    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #144    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #145    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #146    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #147    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #148    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #149    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #150    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #151    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #152    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #153    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #154    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #155    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #156    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #157    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #158    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #159    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #160    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #161    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #162    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #163    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #164    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #165    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #166    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #167    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #168    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #169    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #170    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #171    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #172    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #173    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #174    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #175    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #176    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #177    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #178    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #179    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #180    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #181    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #182    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #183    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #184    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #185    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #186    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #187    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #188    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #189    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #190    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #191    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #192    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #193    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #194    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #195    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #196    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #197    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #198    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #199    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #200    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #201    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #202    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #203    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #204    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #205    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #206    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #207    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #208    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #209    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #210    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #211    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #212    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #213    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #214    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #215    MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4772:32)
flutter: #216    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #217    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #218    _TheatreElement.mount (package:flutter/src/widgets/overlay.dart:493:16)
flutter: #219    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #220    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #221    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #222    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #223    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #224    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #225    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #226    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #227    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #228    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #229    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #230    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #231    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #232    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #233    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #234    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #235    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #236    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #237    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #238    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #239    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #240    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #241    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #242    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #243    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #244    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #245    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #246    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #247    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #248    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #249    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #250    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #251    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #252    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #253    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3782:11)
flutter: #254    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #255    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #256    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #257    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #258    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #259    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #260    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #261    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #262    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #263    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #264    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #265    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #266    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #267    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #268    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #269    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #270    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #271    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #272    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #273    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #274    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #275    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #276    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #277    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #278    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #279    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #280    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #281    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #282    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #283    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #284    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #285    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #286    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #287    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #288    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #289    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #290    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #291    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #292    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3635:5)
flutter: #293    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3630:5)
flutter: #294    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #295    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #296    SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4667:14)
flutter: #297    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2920:14)
flutter: #298    Element.updateChild (package:flutter/src/widgets/framework.dart:2723:12)
flutter: #299    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3666:16)
flutter: #300    Element.rebuild (package:flutter/src/widgets/framework.dart:3508:5)
flutter: #301    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2255:33)
flutter: #302    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:626:20)
flutter: #303    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
flutter: #304    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
flutter: #305    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
flutter: #306    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:751:7)
flutter: #308    _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
flutter: #309    _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
flutter: #310    _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:165:12)
flutter: (elided 3 frames from class _AssertionError and package dart:async)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: A RenderFlex overflowed by 99188 pixels on the bottom.
flutter: Another exception was thrown: 'package:flutter/src/material/bottom_navigation_bar.dart': Failed assertion: line 403 pos 12: 'localizations !=null': is not true.
```

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/19551

https://github.com/flutter/flutter/issues/19551#issuecomment-434112255 explains the problem in detail.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/app.dart [hunk 1]
```diff
@@ -408,10 +412,21 @@ class WidgetsApp extends StatefulWidget {
...
   /// {@template flutter.widgets.widgetsApp.locale}
-  /// The initial locale for this app's [Localizations] widget.
+  /// The initial locale for this app's [Localizations] widget is based
+  /// on this value.
+  ///
+  /// If the 'locale' is null then the system's locale value is used.
   ///
-  /// If the 'locale' is null the system's locale value is used.
+  /// The value of [Localizations.locale] will equal this locale if
+  /// it matches one of the [supportedLocales].
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/app.dart]: This makes me wonder what it will be if it doesn't match... Maybe document that as well?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/app.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/widgets/app.dart]: Good point; updated.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**